src/app/clusters/icd-management-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,120 @@
+/**
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/icd-management-server/ICDManagementCluster.h>
+#include <app/server/Server.h>
+#include <app/static-cluster-config/IcdManagement.h>
+#include <data-model-providers/codegen/ClusterIntegration.h>
+#include <zap-generated/gen_config.h>
+
+namespace {
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::IcdManagement;
+
+static_assert((IcdManagement::StaticApplicationConfig::kFixedClusterConfig.size() == 1 &&
+               IcdManagement::StaticApplicationConfig::kFixedClusterConfig[0].endpointNumber == kRootEndpointId) ||
+              IcdManagement::StaticApplicationConfig::kFixedClusterConfig.size() == 0);
+
+#if CHIP_CONFIG_ENABLE_ICD_CIP
+LazyRegisteredServerCluster<ICDManagementClusterWithCIP> gServer;
+#else
+LazyRegisteredServerCluster<ICDManagementCluster> gServer;
+#endif
+
+constexpr chip::BitMask<OptionalCommands> kEnabledCommands()
+{
+    chip::BitMask<OptionalCommands> result;
+#if defined(ICD_MANAGEMENT_STAY_ACTIVE_REQUEST_COMMAND) || defined(ICD_MANAGEMENT_STAY_ACTIVE_RESPONSE_COMMAND)
+    result.Set(kStayActive);
+#endif
+    return result;
+}
+
+class IntegrationDelegate : public CodegenClusterIntegration::Delegate
+{
+public:
+    ServerClusterRegistration & CreateRegistration(EndpointId endpointId, unsigned clusterInstanceIndex,
+                                                   uint32_t optionalAttributeBits, uint32_t featureMap) override
+    {
+        ICDManagementCluster::OptionalAttributeSet optionalAttributeSet(optionalAttributeBits);
+        const BitMask<OptionalCommands> enabledCommands = kEnabledCommands();
+
+        // Get UserActiveModeTriggerHint
+        BitMask<IcdManagement::UserActiveModeTriggerBitmap> userActiveModeTriggerHint(0);
+        if (Clusters::IcdManagement::Attributes::UserActiveModeTriggerHint::Get(endpointId, &userActiveModeTriggerHint) !=
+            Protocols::InteractionModel::Status::Success)
+        {
+            ChipLogError(Zcl, "Failed to get UserActiveModeTriggerHint, using default (0)");
```

Comment:
Get() does not guarantee anything about the outparam on failure, so if you want it to be 0, set it to 0 here.